go-controller/pkg/ovn/namespace.go

```diff
@@ -218,7 +235,74 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	}
 	defer nsInfo.Unlock()
 
-	annotation := newer.Annotations[hotypes.HybridOverlayExternalGw]
+	var annotation, oldAnnotation string
+	annotation = newer.Annotations[routingExternalGWsAnnotation]
+	oldAnnotation = old.Annotations[routingExternalGWsAnnotation]
+	if annotation != oldAnnotation {
+		var stderr string
+		var err error
+		for podIP, gwToGr := range nsInfo.podExternalRoutes {
+			var mask string
+			for gw, gr := range gwToGr {
+				if utilnet.IsIPv6(net.ParseIP(podIP)) {
+					mask = IPv6FullMask
+				} else {
+					mask = IPv4FullMask
+				}
+				_, stderr, err = util.RunOVNNbctl("--", "--if-exists", "--policy=src-ip",
+					"lr-route-del", gr, podIP+mask, gw)
+				if err != nil {
+					klog.Errorf("Unable to delete src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+				} else {
+					delete(nsInfo.podExternalRoutes, podIP)
+				}
+			}
+		}
+		nsInfo.routingExternalGWs = nil
+		if annotation != "" {
+			for _, v := range strings.Split(annotation, ",") {
```

Comment:
@rcarrillocruz this should be a separate function since it's used here and above in AddNamespace()

Reply:
agreed, done